src/app/server/CommissioningWindowManager.cpp

```diff
@@ -98,11 +103,7 @@ void CommissioningWindowManager::ResetState()
 void CommissioningWindowManager::Cleanup()
 {
     StopAdvertisement(/* aShuttingDown = */ false);
-    DeviceLayer::FailSafeContext & failSafeContext = DeviceLayer::DeviceControlServer::DeviceControlSvr().GetFailSafeContext();
-    if (failSafeContext.IsFailSafeArmed())
-    {
-        failSafeContext.ForceFailSafeTimerExpiry();
-    }
+    ExpireFailSafeIfArmed();
```

Comment:
This does not belong in the CommissioningWindowManager, per spec. We already have to clean that up with the cumulative fail-safe. Since https://github.com/project-chip/connectedhomeip/pull/19582 is merged, we should take this ExpireFailSafeIfArmed out

Reply:
I would rather have that behavior change as a separate PR that we can audit carefully (including checking all the reasons Cleanup is called).  For this PR I just kept existing behavior here.